interlude/src/Features/Score/Graph.fs

```diff
@@ -102,8 +102,40 @@ and ScoreGraph(score_info: ScoreInfo, stats: ScoreScreenStats ref) =
 
     let duration = (score_info.WithMods.LastNote - score_info.WithMods.FirstNote) / score_info.Rate |> format_duration_ms
 
-    let draw_snapshot_info (bounds: Rect) (info: ScoreMetricSnapshot) =
-        Draw.rect bounds Colors.shadow_2.O2
+    do fbo.Unbind()
+
+    override this.Init(parent) =
+        this.Position <- NORMAL_POSITION
+        base.Init parent
+
+    member this.Refresh() = refresh <- true
+
+    member this.ApplyColumnFilter() =
+        stats.Value <- ScoreScreenStats.Generate score_info.Scoring.State.Judgements score_info.Scoring.HitEvents GraphSettings.column_filter
+
+    member this.Keys : int = score_info.WithMods.Keys
+
+    member private this.DrawSnapshotInfo(bounds: Rect, info: ScoreMetricSnapshot) =
+        let outline_thickness = 5.0f 
+    
+        let outline_bounds = bounds.Expand(outline_thickness)
+
+        Draw.rect outline_bounds Colors.white.O4
+        Draw.rect bounds Colors.shadow_2.O4
+        let mouse_x, mouse_y = Mouse.pos()
+        let percent = (mouse_x - bounds.Left) / bounds.Width
+        let adjusted_x = 
+            max bounds.Left (min mouse_x (bounds.Right - outline_thickness / 2.0f))
```

Comment:
Bonus info you may find interesting:
There is a built in function for "clamping" a number between two values:

`System.Math.Clamp(value, min, max)`

I rarely use it because:
- If you give it an upper bound that is *lower* than the lower bound, the game will crash, and mistakes can happen :)
`min`ning and then `max`ing will never crash and returns the lower bound if the upper bound is below it
- In F# you can write `<some calculation> |> min <max value> |> max <min value>` as a suffix which I like typing and reading more than wrapping things in commas and brackets

It's still used in some places where the lower and upper bounds are constant values or it's definitely impossible for them to conflict